### ISSUE TITLE
'to_meter' on 'proj=ob_tran o_proj=latlon' broke in 6.1.1

### ISSUE THREAD

**Issue by QuLogic** [OSGeo/PROJ#1782]
In Cartopy, there is a `RotatedGeodetic` CRS which uses 'ob_tran' to define a rotated coordinate system, using degrees with 'o_proj=latlon'. It specifies `to_meter=math.radians(1)` to produce results in degrees as well. Cartopy has not been updated to use any new API, so a transform from `PlateCarree` to `RotatedGeodetic` is something like:

```
pj_transform("+ellps=WGS84 +a=57.29577951308232 +proj=eqc +lon_0=0.0 +no_defs"
             "+ellps=WGS84 +proj=ob_tran +o_proj=latlon +o_lon_p=0.0 +o_lat_p=90.0 +lon_0=360.0 +to_meter=0.0174532925199433 +no_defs",
             nx, ny, &x, &y, NULL);
```
(I am skipping all the init stuff here, of course.)

The pole location and rotation angle _should_ result in no change to the result. Starting with 6.1.1, the `to_meter` has no effect and results are `to_meter` times smaller than they should be (or the inverse, if transforming in reverse.) For example, with 6.1.0, some test points are:
```
(-179, -89) -> (-179.00000000000003, -89.00000000000007)
(-179, 0) -> (-179.00000000000003, 3.507820309861257e-15)
(-179, 89) -> (-179.00000000000003, 89.00000000000007)
(0, -89) -> (0.0, -89.00000000000007)
(0, 0) -> (0.0, -3.508354649267437e-15)
(0, 89) -> (0.0, 89.00000000000007)
(179, -89) -> (178.99999999999997, -89.00000000000007)
(179, 0) -> (178.99999999999997, 3.507820309861257e-15)
(179, 89) -> (178.99999999999997, 89.00000000000007)
```
but with 6.1.1, the results are:
```
(-179, -89) -> (-3.124139361069851, -1.553343034274955)
(-179, 0) -> (-3.124139361069851, 6.122301397540666e-17)
(-179, 89) -> (-3.124139361069851, 1.553343034274955)
(0, -89) -> (0.0, -1.553343034274955)
(0, 0) -> (0.0, -6.123233995736766e-17)
(0, 89) -> (0.0, 1.553343034274955)
(179, -89) -> (3.12413936106985, -1.553343034274955)
(179, 0) -> (3.12413936106985, 6.122301397540666e-17)
(179, 89) -> (3.12413936106985, 1.553343034274955)
```

I can also reproduce this directly with `proj`, 6.1.0:
```
for lon in -179 0 179; do
 for lat in -89 0 89; do
  echo $lon $lat
 done
done | proj -E +ellps=WGS84 +proj=ob_tran +o_proj=latlon +o_lon_p=0.0 +o_lat_p=90.0 +lon_0=360.0 +to_meter=0.0174532925199433 +no_defs

-179 -89        -179.00 -89.00
-179 0  -179.00 0.00
-179 89 -179.00 89.00
0 -89   0.00    -89.00
0 0     0.00    -0.00
0 89    0.00    89.00
179 -89 179.00  -89.00
179 0   179.00  0.00
179 89  179.00  89.00
```
vs 6.1.1:
```
for lon in -179 0 179; do
 for lat in -89 0 89; do
  echo $lon $lat
 done
done | proj -E +ellps=WGS84 +proj=ob_tran +o_proj=latlon +o_lon_p=0.0 +o_lat_p=90.0 +lon_0=360.0 +to_meter=0.0174532925199433 +no_defs

-179 -89        -3.12   -1.55
-179 0  -3.12   0.00
-179 89 -3.12   1.55
0 -89   0.00    -1.55
0 0     0.00    -0.00
0 89    0.00    1.55
179 -89 3.12    -1.55
179 0   3.12    0.00
179 89  3.12    1.55
```

Bisect says this started with 11a53b7c0255087c2b324b36147deb837902dfb9.

**Comment by snowman2** [comment]
How does it behave in 6.2? Likely related to #1601

**Comment by QuLogic** [comment]
I saw that, but it is not fixed as of `master` (29474bd7fdac013841a4795e93f714bba7791c56).

**Comment by kbevers** [comment]
This does indeed look problematic. The fix for https://github.com/OSGeo/PROJ/issues/1525 introduced this as a side effect.
